helix-term/src/application.rs

```diff
@@ -254,9 +251,19 @@ impl Application {
         use helix_view::graphics::Rect;
         match signal {
             signal::SIGTSTP => {
+                use nix::{
+                    sys::signal::{self, Signal},
+                    unistd::Pid,
+                };
                 self.compositor.save_cursor();
                 self.restore_term().unwrap();
```

Comment:
It seems to me that you didn't follow the signal-hook docs: you're supposed to call `low_level::raise(SIGSTOP)?;`, not emulate_default_handler: https://docs.rs/signal-hook/latest/signal_hook/#a-complex-signal-handling-with-a-background-thread

Reply:
Hmm, seems to be the same underlying call though

Reply:
Yes, it's the same thing.